src/webviews/extension-side/dataviewer/dataViewerDependencyService.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import { SemVer } from 'semver';
+import { ProductNames } from '../../../kernels/installer/productNames';
+import { Product } from '../../../kernels/installer/types';
+import { IApplicationShell } from '../../../platform/common/application/types';
+import { traceWarning } from '../../../platform/logging';
+import { IsCodeSpace } from '../../../platform/common/types';
+import { parseSemVer } from '../../../platform/common/utils';
+import { DataScience, Common } from '../../../platform/common/utils/localize';
+import { EnvironmentType } from '../../../platform/pythonEnvironments/info';
+import { sendTelemetryEvent, Telemetry } from '../../../telemetry';
+import { IDataViewerDependencyService } from './types';
+import { IKernel } from '../../../kernels/types';
+import { executeSilently } from '../../../kernels/helpers';
+
+export const minimumSupportedPandaVersion = '0.20.0';
+export const getVersionOfPandasCommand =
+    'import pandas as _VSCODE_pandas;print(_VSCODE_pandas.__version__);del _VSCODE_pandas';
+
+function isVersionOfPandaSupported(version: SemVer) {
+    return version.compare(minimumSupportedPandaVersion) > 0;
+}
+
+/**
+ * Responsible for managing dependencies of a Data Viewer.
+ */
+@injectable()
+export class DataViewerDependencyService implements IDataViewerDependencyService {
+    constructor(
+        @inject(IApplicationShell) private readonly applicationShell: IApplicationShell,
+        @inject(IsCodeSpace) private isCodeSpace: boolean
+    ) {}
+
+    private packaging(kernel: IKernel): 'pip' | 'conda' {
```

Comment:
There are a lot of different env types here. Might not be safe to just call everything not conda as pip. As a first pass we could just ignore the other types. Poetry and what not.

Reply:
Does this still apply with `%pip` and `%conda`?

Reply:
I think that's what we're doing 🤔 `isConda ? 'conda' : 'pip'`